File: carcassonne-master/src/left_side.cc
#include "left_side.h"

#include "left_side_connections.h"

namespace carcassonne
{

LeftSide::LeftSide() : Side()
{
	middle_.reset(new TerrainSegment);
	connections_.reset(new LeftSideConnections);
	
}

LeftSide::LeftSide(const LeftSide& src) : Side(src)
{
	CopyFrom(src);	
}

LeftSide& LeftSide::
	operator=(const LeftSide& rhs)
{
	if(this == &rhs) {
		return (*this);
	}
	
	CopyFrom(rhs);
	
	return (*this);
}
 	
void LeftSide::
	CopyFrom(const LeftSide& src)
{
}
 
Side* LeftSide::
	Clone() const
{
	return new LeftSide(*this);
}

void LeftSide::ConnectedToTop()
{
	connections_->set_clockwise(true);
	
}

void LeftSide::ConnectedToRight()
{
	connections_->set_across(true);
	
}

void LeftSide::ConnectedToBottom()
{
	connections_->set_counterclockwise(true);
	
}

void LeftSide::ConnectedToLeft()
{
}

bool LeftSide::IsConnectedToTop()
{
  return connections_->clockwise();
  
}

bool LeftSide::IsConnectedToRight()
{
  return connections_->across();
  
}

bool LeftSide::IsConnectedToBottom()
{
  return connections_->counterclockwise();
  
}

bool LeftSide::IsConnectedToLeft()
{
  return true;
  
}

std::string LeftSide::ToString() const
{
  return "left side with " + connections_->ToString();
  
}

LeftSide::~LeftSide()
{
}

}

File: js/LeftSide.js
/**
 * Created by marcogobbi on 10/10/2017.
 */
import Side from "./Side";
export default class LeftSide extends Side {
    constructor(src) {
        super(src);
        this.middle_ = new TerrainSegment();
        this.connections_ = new LeftSideConnections();
    }

    Clone() {
        return new LeftSide(this);
    }

    ConnectedToTop() {
        this.connections_.set_clockwise(true);

    }

    ConnectedToRight() {
        this.connections_.set_across(true);

    }

    ConnectedToBottom() {
        this.connections_.set_counterclockwise(true);

    }

    ConnectedToLeft() {
    }

    IsConnectedToTop() {
        return this.connections_.clockwise();

    }

    IsConnectedToRight() {
        return this.connections_.across();

    }

    IsConnectedToBottom() {
        return this.connections_.counterclockwise();

    }

    IsConnectedToLeft() {
        return true;

    }

    ToString() {
        return "left side with " + this.connections_.ToString();

    }
}